vit-servicing-station-tests/src/tests/rest/funds.rs

```diff
@@ -58,3 +58,17 @@ pub fn get_funds_by_id() -> Result<(), Box<dyn std::error::Error>> {
 
     Ok(())
 }
+
+// set the internal id of each challenge consistently
+fn normalize(fund: &mut Fund) {
+    fund.challenges.sort_by_key(|fund| fund.id);
+    fund.chain_vote_plans.sort_by_key(|c| c.id);
+    fund.chain_vote_plans.sort_by_key(|c| c.id);
+
+    let mut id = 0;
+    #[allow(clippy::explicit_counter_loop)]
+    for challenge in &mut fund.challenges {
+        challenge.internal_id = id;
+        id += 1;
+    }
+}
```

Comment:
It seemed like the `internal_id`s were wrong, not jut that the challenges were out of order, though maybe I'm misunderstanding.

I'm also wondering whether the `internal_id` field needs to exist. The name `internal_id` makes me think it's not needed by the app, though maybe that's wrong?

Reply:
Hmm, I guess this needs a bit more investigation. For reference, this is an example failure from CI:
```
challenges: [
         Challenge {
             internal_id: 1,
<            id: 1699916299,
<            challenge_type: Simple,
<            title: "Front-line discrete internet solution",
<            description: "Enhanced",
<            rewards_total: 4439,
<            proposers_rewards: 8918,
<            fund_id: 299688433,
<            challenge_url: "http://kovacek-and-sons.net/",
<            highlights: None,
<        },
<        Challenge {
<            internal_id: 2,
             id: 476497144,
             challenge_type: CommunityChoice,
             title: "Progressive homogeneous middleware",
             description: "Balanced",
             rewards_total: 2810,
             proposers_rewards: 1175,
             fund_id: 299688433,
             challenge_url: "http://ruecker-and-hyatt-group.net/",
             highlights: Some(
                 ChallengeHighlights {
                     sponsor: "Reichert and Corwin and Sons",
                 },
             ),
         },
>        Challenge {
>            internal_id: 2,
>            id: 1699916299,
>            challenge_type: Simple,
>            title: "Front-line discrete internet solution",
>            description: "Enhanced",
>            rewards_total: 4439,
>            proposers_rewards: 8918,
>            fund_id: 299688433,
>            challenge_url: "http://kovacek-and-sons.net/",
>            highlights: None,
>        },
     ],
```
Seems like the `internal_id`s are swapped compared to what the assert expects. I'll see if I can work out what's going on here

Reply:
I think it could simply depend on the order in which challenges are inserted at lines 30/31, IIRC `internal_id` follows insertion order. I would anyway try to get it right so that we don't arbitrarily overwrite returned responses from the api

Reply:
Think I've got this sorted, it seems to come from the fact that `internal_id` is ignored when inserting into the DB, ended up being simpler than I thought